machine_learning_hep/processer.py

```diff
@@ -181,9 +182,14 @@ def __init__(self, case, datap, run_param, mcordata, p_maxfiles, # pylint: disab
         if os.path.isdir(self.d_root):
             self.l_path = list_folders(self.d_root, self.n_root, self.p_maxfiles,
                                        self.select_jobs)
-        else:
+        elif is_file_indir(self.d_pkl, self.n_reco):
```

Comment:
Why not simply this?
```suggestion
        elif glob.glob(f"{self.d_pkl}/**/{self.n_reco}", recursive=True):
```